src/mbed_tools/project/_internal/libraries.py

```diff
@@ -64,19 +64,19 @@ def resolve(self) -> None:
             repo = git_utils.clone(git_ref.repo_url, lib.source_code_path)
             if git_ref.ref:
                 logger.info(f"Checking out revision {git_ref.ref} for library {git_ref.repo_url}.")
-                git_utils.checkout(repo, git_ref.ref)
+                git_utils.deploy(repo, git_ref.ref)
 
         # Check if we find any new references after cloning dependencies.
         if list(self.iter_unresolved()):
             self.resolve()
 
-    def checkout(self, force: bool) -> None:
+    def deploy(self, force: bool) -> None:
```

Comment:
This is an internal API, I wouldn't bother changing the name here.

Reply:
Idea is to keep internal and external in sync to avoid confusion

Reply:
I don't think it is confusing. Internal APIs should be describing "how we are doing it", and we are doing the deploy by using git checkout. We should just change the command names and the click callback names as they are really the only place the "deploy" terminology makes sense.

Reply:
I'm not clear on your thoughts here, please elaborate with an example.